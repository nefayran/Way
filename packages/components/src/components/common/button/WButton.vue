<template>
  <button
    :class="[classes.button]"
    @click="$emit('click', $event)"
    @mouseover="hover"
  >
    <span :class="[classes.text]">
      <slot />
    </span>
  </button>
</template>

<script lang="ts" setup>
import { defineProps, withDefaults, computed, defineEmits } from "vue";

interface Props {
  theme: string;
  color?: string;
  fill?: string;
  size?: string;
}

const props = withDefaults(defineProps<Props>(), {
  theme: "light",
  color: "white--base",
  fill: "indigo--base",
  size: "lg",
});

const emit = defineEmits(["click"]);

const classes = computed(() => {
  return {
    button: `w-button w-button--${props.size} ${props.fill}`,
    text: `w-text text-${props.color}`,
  };
});

const hover = (event: any) => {
  console.log("hover");
};
</script>

<style lang="scss">
/* === Button Sizes === */
$sizes: (xl, lg, md, sm, xs) !default;
.w-button {
  border: none;

  @each $size in $sizes {
    &--#{$size} {
      height: var(--height-#{$size});
      min-height: var(--height-#{$size});
      padding-left: var(--space-#{$size});
      padding-right: var(--space-#{$size});
      border-radius: var(--border-radius-#{$size});
      min-width: var(--width-#{$size});

      .w-text {
        font-size: var(--font-size-#{$size});
      }
    }
  }

  &:hover {
    cursor: pointer;
  }
}
</style>
