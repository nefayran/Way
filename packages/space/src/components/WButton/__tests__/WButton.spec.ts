import { mount } from "@vue/test-utils";
import WButton from "../WButton.vue";

describe("WButton.vue", () => {
  test("should have an w-button class", () => {
    const wrapper = mount(WButton, {
      propsData: {
        theme: "light",
      },
    });
    expect(wrapper.classes()).toContain("w-button");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
