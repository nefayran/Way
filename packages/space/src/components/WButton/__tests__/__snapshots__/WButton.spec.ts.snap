// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WButton.vue should have an w-button class 1`] = `<button class="w-button w-button--lg secondary--base"><span class="w-text font-secondary text-primary--base">@way</span></button>`;
